src/richie/apps/core/context_processors.py

```diff
@@ -86,3 +92,98 @@ def site_metas(request):
 
     context["FRONTEND_CONTEXT"] = json.dumps(context["FRONTEND_CONTEXT"])
     return context
+
+
+class WebAnalyticsContextProcessor:
+    """
+    Context processor to add Web Analytics tracking information to Richie CMS templates and
+    frontend.
+    """
+
+    def context_processor(self, request: HttpRequest) -> dict:
+        """
+        Real implementation of the context processor for the Web Analytics core app sub-module
+        """
+        context = {}
+        if hasattr(request, "current_page"):
+            # load web analytics settings to the context
+            if getattr(settings, "WEB_ANALYTICS_ID", None):
+                context["WEB_ANALYTICS_ID"] = settings.WEB_ANALYTICS_ID
+            context["WEB_ANALYTICS_LOCATION"] = getattr(
+                settings, "WEB_ANALYTICS_LOCATION", "head"
+            )
+            context["WEB_ANALYTICS_PROVIDER"] = getattr(
+                settings, "WEB_ANALYTICS_PROVIDER", "google_analytics"
+            )
+            context["WEB_ANALYTICS_DIMENSIONS"] = (
+                self.get_dimensions(request)
+                if getattr(settings, "WEB_ANALYTICS_ID", None)
+                else None
+            )
+        return context
+
+    # pylint: disable=no-self-use
+    def is_load_dimensions(self, request: HttpRequest) -> bool:
+        """
+        Method to easily enable or disable the load of web analytics dimensions
+        """
+        return getattr(settings, "WEB_ANALYTICS_ID", None)
+
```

Comment:
What I meant by "making it simpler" was that this method can now be removed. If you don't like the inline if, keep it on 2 lines. 
```suggestion
```

Reply:
Ok, now I understand what you have wroten. I'm going to remove that method and use a "normal" if.

```python
            if getattr(settings, "WEB_ANALYTICS_ID", None):
                context["WEB_ANALYTICS_DIMENSIONS"] = self.get_dimensions(request)
```